### PR TITLE
Handle invalid request token operation values

### PR DESCRIPTION
**Pull request by ncounter**
Fix https://github.com/openSUSE/open-build-service/issues/14810

tl;dr
Handle and reject requests when `token`'s `operation` values are not valid.

Long version:
The former issue #14810 hits this scenario:
- user wants to create an `rss_token`
- he finds `token`'s endpoints to use
- he submit a `create token` request with an `operation` parameter value which is meant to be for an `rss` token
- eventually the `create token` allows any `operation` value in a fallback creating a `service token`, which is far away from the user's goal

The problem can be solved by two different actions:
1. document the `rss_token` creation (because at the moment I couldn't find it)
2. prevent creating a `token` if the `operation` value has no meaningful value (AKA only if it matches one of the possible values or it is completely omitted, and it fails in case of _any other value_ otherwise), reject the request and hints the user with a message about what happened

This PR addresses the second point.

### How to test this PR
- checkout this branch
- hit the `POST /person/$login/token?cmd=create&operation=XYZ` multiple times trying to change the `operation=` parameter value with: 
  - one of the possible API values `['rebuild', 'release', 'runservice', 'workflow']` --> it creates a token accordingly
  - omitting the `operation` parameter completely --> it creates a `runservice` token
  - using `operation=rss` --> it fails with an error message
  - using `operation=blabla` --> it fails with an error message

### CODE REVIEW

File: src/api/app/controllers/person/token_controller.rb [hunk 1]
```diff
@@ -50,5 +51,17 @@ def set_package
 
       @package = Package.get_by_project_and_name(params[:project], params[:package])
     end
+
+    def validate_operation
+      operation_param = params[:operation]
+      return if operation_param.nil? || Token::OPERATIONS.include?(operation_param.capitalize)
```
Comment (eduardoj) [inline src/api/app/controllers/person/token_controller.rb:57]: Is this line changing from the allowed token type `runservice` to `service`? :thinking:

This is already documented [here](https://github.com/openSUSE/open-build-service/blob/221811444256005c46b48af45f7f6313496c8499/src/api/public/apidocs/paths/person_login_token.yaml#L89).

Comment (ncounter) [reply src/api/app/controllers/person/token_controller.rb:57]: Ah, I think I misused https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/token.rb#L27 set of possible values instead. But now I'm puzzled: why the API uses a different set than the webUI?

Comment (ncounter) [reply src/api/app/controllers/person/token_controller.rb]: I added a change to handle and accept both webUI and API values keeping the same behavior otherwise.

https://github.com/openSUSE/open-build-service/pull/14979/files#diff-67a22ba8cf70b37462f7ef3e9fe0370283ffae0f2ace759387591b1bbb06324dR57-R62

But it still feels dirty and hacked somehow, so we should rather align the two usages eventually.

File: src/api/app/controllers/person/token_controller.rb [hunk 2]
```diff
@@ -50,5 +51,18 @@ def set_package
...
       @package = Package.get_by_project_and_name(params[:project], params[:package])
     end
+
+    def validate_operation
+      operation_param = params[:operation]
+      # TODO: align operation parameter allowed values
+      # - webUI: https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/token.rb#L27
+      # - API: https://github.com/openSUSE/open-build-service/blob/master/src/api/public/apidocs/paths/person_login_token.yaml#L89
+      return if operation_param.nil? ||
+                ['runservice', 'rebuild', 'release'].include?(operation_param) # possible API parameter values
```
Comment (eduardoj) [inline src/api/app/controllers/person/token_controller.rb:61]:
```suggestion
                ['runservice', 'rebuild', 'release', 'workflow'].include?(operation_param) # possible API parameter values
```
We also support the `workflow` value for the `operation` parameter, as the failing test pointed out. It is not yet documented, but we support it.

**Review by eduardoj — APPROVED** [review]
